lib/lightning/projects.ex

```diff
@@ -245,6 +245,20 @@ defmodule Lightning.Projects do
     end)
   end
 
+  @spec project_retention_policy_for(Attempt.t()) ::
+          :retain_all | :erase_all | :erase_successful_only
```

Comment:
Can we move these into a type that is defined elsewhere, so any changes will propagate to any other functions?